components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -521,7 +521,11 @@ void RewardsServiceImpl::Shutdown() {
     }
   }
 
+  for (const auto fetcher : fetchers_) {
+    delete fetcher.first;
+  }
   fetchers_.clear();
```

Comment:
is this line needed then based on the fact that you delete all fetchers in for above?

Reply:
this clears out the pointers which point to invalid memory after they were deleted.